src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheWarmer.php

```diff
@@ -56,6 +61,8 @@ public function warmUp(string $cacheDir, ?string $buildDir = null): array
             }
         }
 
+        $this->twig->setCache($originalCache);
```

Comment:
I was about to suggest wrapping this reset statement in a `finally` block. But this is not necessary since any exception is already catched.

Reply:
Only `Twig\Error\Error` is caught

Reply:
@Okhoshi can you please move this reset to a finally block ?